lib/practice_exercises.rb

```diff
@@ -1,13 +1,49 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: log(n)
+# Space Complexity: O(1)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  string.downcase!
+  string.gsub!(/[^A-Za-z]/, '')
+  (string.length / 2).times do |i|
+    return false if string[i] != string[string.length - 1 - i]
+  end
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
-def longest_prefix(strings)
-  raise NotImplementedError, "Not implemented yet"
+# Time Complexity: mlog(n)
+# Space Complexity: O(n)
+# I tried to come up with a binary, couldn't get my solution to work, ended up
+# finding this solution online: https://leetcode.com/problems/longest-common-prefix/discuss/292306/ruby-100-faster-than-ruby-(binary-search)
+# tweeked to get better understanding and to reduce time complexity
+def common_prefix?(strings, string1, low, mid)
```

Comment:
That's an interesting solution searching for a common prefix with binary search.  

I'd even say it's O(n * log m) where n is the number of strings and m is the length of the shortest string.